py/objbool.c

```diff
@@ -83,5 +83,5 @@ const mp_obj_type_t mp_type_bool = {
     .binary_op = bool_binary_op,
 };
 
-const mp_obj_bool_t mp_const_false_obj = {{&mp_type_bool}, false};
-const mp_obj_bool_t mp_const_true_obj = {{&mp_type_bool}, true};
+const mp_obj_bool_t MP_SMALL_DATA_SECTION(".rodata.mp_const_false_obj") mp_const_false_obj = {{&mp_type_bool}, false};
+const mp_obj_bool_t MP_SMALL_DATA_SECTION(".rodata.mp_const_true_obj") mp_const_true_obj = {{&mp_type_bool}, true};
```

Comment:
Are these attributes needed?  Can't the linker script access these entities anyway just by their name?

Reply:
Currently they are not needed, since that's exactly the name selected by `-fdata-sections`. I thought it's better be explicit, also in case that flag is ever removed. We can just remove those section definitions though.

Reply:
I think it's fair enough to require a port to use `-fdata-sections` to use this feature, in which case these bits can go, to not clutter the core (maybe in the future if this feature turns out to be really useful and these need to be added they can be added then).

Reply:
:+1: , removing.